internal/commands/app/create_inputs_test.go

```diff
@@ -372,111 +373,659 @@ func TestAppCreateInputsResolveDirectory(t *testing.T) {
+		_, ui := mock.NewUI()
 		var expectedGroupID string
 		ac := mock.AtlasClient{}
-		ac.DataLakesFn = func(groupID string) ([]atlas.DataLake, error) {
+		ac.DatalakesFn = func(groupID string) ([]atlas.Datalake, error) {
 			expectedGroupID = groupID
-			return nil, nil
+			return []atlas.Datalake{
+				{Name: "test-datalake-1"},
+				{Name: "test-datalake-2"},
+			}, nil
 		}
 
-		inputs := createInputs{DataLake: "test-datalake"}
+		inputs := createInputs{
+			newAppInputs:         newAppInputs{Name: "test-app"},
+			Datalakes:            []string{"test-datalake-1", "test-datalake-2"},
+			DatalakeServiceNames: []string{"mongodb-datalake", "another-data-source"},
+		}
 
-		_, err := inputs.resolveDataLake(ac, "123")
-		assert.Equal(t, errors.New("failed to find Atlas data lake"), err)
+		ds, _, err := inputs.resolveDatalakes(ui, ac, "123")
+		assert.Nil(t, err)
+
+		assert.Equal(t, []dataSourceDatalake{
+			{
+				Name: "mongodb-datalake",
+				Type: realm.ServiceTypeDatalake,
+				Config: configDatalake{
+					DatalakeName: "test-datalake-1",
+				},
+			},
+			{
+				Name: "another-data-source",
+				Type: realm.ServiceTypeDatalake,
+				Config: configDatalake{
+					DatalakeName: "test-datalake-2",
+				},
+			},
+		}, ds)
 		assert.Equal(t, "123", expectedGroupID)
 	})
 
+	t.Run("should warn user if data lakes are not found and auto confirm is set", func(t *testing.T) {
+		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+		defer teardown()
+
+		console, ui, err := mock.NewConsoleWithOptions(mock.UIOptions{AutoConfirm: true})
+		assert.Nil(t, err)
+		defer console.Close()
+
+		testApp := realm.App{
+			ID:          primitive.NewObjectID().Hex(),
+			GroupID:     primitive.NewObjectID().Hex(),
+			ClientAppID: "test-app-abcde",
+			Name:        "test-app",
+		}
+
+		rc := mock.RealmClient{}
+		rc.CreateAppFn = func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
+			return testApp, nil
+		}
+		rc.ImportFn = func(groupID, appID string, appData interface{}) error {
+			return nil
+		}
+		rc.TemplatesFn = func() ([]realm.Template, error) {
+			return []realm.Template{}, nil
+		}
+
+		ac := mock.AtlasClient{}
+		ac.DatalakesFn = func(groupID string) ([]atlas.Datalake, error) {
+			return []atlas.Datalake{
+				{Name: "test-datalake-1"},
+			}, nil
+		}
+
+		dummyDatalakes := []string{"test-dummy-lake-1", "test-dummy-lake-2"}
+		inputs := createInputs{
+			newAppInputs:         newAppInputs{Name: testApp.Name, Project: testApp.GroupID},
+			Datalakes:            []string{"test-datalake-1", dummyDatalakes[0], dummyDatalakes[1]},
+			DatalakeServiceNames: []string{"mongodb-datalake"},
+		}
+
+		doneCh := make(chan (struct{}))
+		go func() {
+			defer close(doneCh)
+			console.ExpectString(fmt.Sprintf("Note: The following data sources were not linked because they could not be found: %s", strings.Join(dummyDatalakes[:], ", ")))
```

Comment:
usual comment on `[:]` here and opting to just write the string out if you feel it makes things clearer